src/main/kotlin/io/github/nomisrev/routes/articles.kt

```diff
@@ -52,3 +74,27 @@ private object OffsetDateTimeIso8601Serializer : KSerializer<OffsetDateTime> {
     encoder.encodeString(value.toString())
   }
 }
+
+fun Application.articleRoutes(articleService: ArticleService, jwtService: JwtService) = routing {
+  route("/articles") {
+    post {
```

Comment:
I have some questions:

1 - Since there's no `articleId` on the [specification](https://github.com/gothinkster/realworld/blob/0ef2ba3996714c76e8f1a06cdd9895c4c3d8cd9a/api/openapi.yml#L497), how should we proceed with it here?
2 - how should we implement validations?

Reply:
1. I think we need to add an `ArticleResponse` that matches the specification. So we have 3 models, the one matching the OpenAPI, the one used in our business logic domain, and the one used in our database. The database model is generated by SqlDelight, and is typically ignored by using `executeOne { /** transform properties to  business logic domain */ }`.

2. You can check `validations.kt` for validation, I think it already includes all validation checks. `body`, `description`, `title` and `tags`. So something like:

```kotlin
fun NewArticle.validate(): Either<IncorrectInput, NewArticle> =
  zipOrAccumulate(
      title.validTitle(),
      description.validDescription(),
      body.validBody(),
      validTags(tags)
      :: NewArticle
    )
    .mapLeft(::IncorrectInput)
```